File: Courses/PHP Web Basics - май 2019/18. Exercise Polymorphism, Interfaces and Abstraction/01.03.Ferrari/Ferrari.php
<?php

require_once "CarInterface.php";
class Ferrari implements CarInterface
{
    /**
     * @var string
     */
    private $driverName;
    /**
     * @var string
     */
    private $model;

    /**
     * Ferrari constructor.
     * @param string $driverName
     * @param string $model
     */
    public function __construct(string $driverName, string $model = "488-Spider")
    {
        $this->driverName = $driverName;
        $this->model = $model;
    }

    /**
     * @return string
     */
    public function getDriverName(): string
    {
        return $this->driverName;
    }


    public function brakes()
    {
        return "Brakes!";
    }

    public function gas()
    {
        return "Zadu6avam sA!";
    }

    static function forUrl(string $str, string $rep="-")
    {
        return  $str = str_replace (" ", $rep, $str);
    }

    function __toString()
    {
        return "{$this->model}/" . $this->brakes() . "/" . $this->gas() . "/{$this->driverName}";
    }
}

File: Courses/C# OOP/Basics/12. Interfaces and Abstraction Exercise/Interfaces and Abstraction - Exercise/03.Ferrari/Ferrari.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _03.Ferrari
{
    public class Ferrari : ICar
    {

        public Ferrari(string driver)
        {
            Driver = driver;
        }

        public string Model => "488-Spider";

        public string Driver { get; set; }

        public string Brakes()
        {
            return "Brakes!";
        }

        public string Gas()
        {
            return "Zadu6avam sA!";
        }
        public override string ToString()
        {
            return $"{Model}/{Brakes()}/{Gas()}/{Driver}";
        }
    }
}
